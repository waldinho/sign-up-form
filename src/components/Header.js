import React from 'react';
import styled from 'styled-components';

const Header = () => {

    return (
        <Wrapper>
            <header className='App-header'>
                <img className='logo' src='https://www.southerncrossaustereo.com.au:443/media/1973/logo-sca.svg' alt='SCA' title='SCA'/>
                <h3>Subscribe</h3>
            </header>
        </Wrapper>
    )
}

const Wrapper = styled.div`
.App-header {
    background-color: #333;
    min-height: 10vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-size: calc(10px + 2vmin);
    color: white;
    min-width: 335px;
    padding: 30px 0 0 0;
    position: fixed;
    top: 0;
    width: -webkit-fill-available;
    z-index: 2;
    .logo {
        width: 120px;
    }
}
`

export default Header