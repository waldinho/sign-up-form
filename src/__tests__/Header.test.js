import React from 'react';
import ReactDOM from 'react-dom';
import Header from '../components/Header';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Header />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('<Header /> with no props', () => {
    const container = shallow(<Header />);
    it('should match the snapshot', () => {
      expect(container.html()).toMatchSnapshot();
      expect(container.find('img').prop('src')).toEqual('https://www.southerncrossaustereo.com.au:443/media/1973/logo-sca.svg');
      expect(container.find('img').prop('alt')).toEqual('SCA');
      expect(container.find('img').prop('title')).toEqual('SCA');
      expect(container.find('img').prop('className')).toEqual('logo');
    })
})
