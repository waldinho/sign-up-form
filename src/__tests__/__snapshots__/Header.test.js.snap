// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> with no props should match the snapshot 1`] = `"<div class=\\"sc-bdVaJa hODwFO\\"><header class=\\"App-header\\"><img class=\\"logo\\" src=\\"https://www.southerncrossaustereo.com.au:443/media/1973/logo-sca.svg\\" alt=\\"SCA\\" title=\\"SCA\\"/><h3>Subscribe</h3></header></div>"`;
